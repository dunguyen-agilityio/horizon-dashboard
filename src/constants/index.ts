export * from './common';
export * from './images';
export * from './messages';
export * from './routes';
export * from './text';
export * from './endpoint';
export * from './session';
