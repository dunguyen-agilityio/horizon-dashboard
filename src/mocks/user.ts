import { TImage } from '@/models/Image';
import { TUser } from '@/models/User';
import { StrapiResponse } from '@/types/strapi';
import { UserResponse1 } from '@/utils/nft';

export const MOCK_AVATAR: TImage = {
  hash: 'image_de1e29b0b7',
  url: 'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1726630608/image_de1e29b0b7.png',
};

export const MOCK_USERS: TUser[] = [
  {
    id: '1',
    username: 'Adela Parkson',
    firstName: 'User 1',
    email: 'user1@gmail.com',
    role: { name: 'Creative Director' },
    createdAt: new Date('2024-08-17T07:25:04.188Z'),
    rating: 29,
    avatar: MOCK_AVATAR,
  },
  {
    id: '2',
    username: 'user2',
    firstName: 'User 2',
    email: 'user2@gmail.com',
    role: { name: 'Product Designer' },
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
    rating: 72,
    avatar: MOCK_AVATAR,
  },
  {
    id: '3',
    username: 'user3',
    firstName: 'User 3',
    email: 'user3@gmail.com',
    role: { name: 'Product Designer' },
    createdAt: new Date('2024-01-13T07:25:04.188Z'),
    rating: 89,
    avatar: MOCK_AVATAR,
  },
  {
    id: '4',
    username: 'user4',
    firstName: 'User 4',
    email: 'user4@gmail.com',
    role: { name: 'Junior Graphic Designer' },
    createdAt: new Date('2024-08-18T07:25:04.188Z'),
    rating: 52,
    avatar: MOCK_AVATAR,
  },
];

export const MOCK_ROLE_RESPONSE: StrapiResponse<{ id: string; name: string }> =
  {
    id: '1',
    attributes: { name: 'Developer' },
  };

export const MOCK_IMAGE_RESPONSE: StrapiResponse<TImage> = {
  id: '1',
  attributes: {
    hash: 'image_de1e29b0b7',
    url: 'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1726630608/image_de1e29b0b7.png',
  },
};

export const MOCK_USERS_RESPONSE: StrapiResponse<UserResponse1>[] =
  MOCK_USERS.map(({ id, ...attributes }) => ({
    attributes: {
      ...attributes,
      avatar: { data: MOCK_IMAGE_RESPONSE },
      role: MOCK_ROLE_RESPONSE,
    },
    id,
  }));
