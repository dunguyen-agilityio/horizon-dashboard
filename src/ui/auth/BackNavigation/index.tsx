'use client';

// Libs
import Link from 'next/link';
import { usePathname } from 'next/navigation';

// Components
import { ArrowRight } from '@/icons';
import { BoxIcon, Text } from '@/components';

// Constants
import {
  AUTH_NAVIGATION_HEADER,
  AUTH_ROUTES,
  PUBLIC_ROUTES,
} from '@/constants';

// Types
import { TEXT_VARIANT } from '@/types/text';

const BackNavigation = () => {
  const pathname = usePathname();

  const getNavigationHeader = (): { title: string; href: string } => {
    switch (true) {
      case pathname.includes(AUTH_ROUTES.SIGN_IN):
        return {
          title: AUTH_NAVIGATION_HEADER[PUBLIC_ROUTES.DASHBOARD],
          href: PUBLIC_ROUTES.DASHBOARD,
        };

      case pathname.includes(AUTH_ROUTES.FORGET_PASSWORD):
      case pathname.includes(AUTH_ROUTES.SIGN_UP):
      case pathname.includes(AUTH_ROUTES.RESET_PASSWORD):
        return {
          title: AUTH_NAVIGATION_HEADER[AUTH_ROUTES.SIGN_IN],
          href: AUTH_ROUTES.SIGN_IN,
        };

      default:
        return { title: '', href: '' };
    }
  };

  const { title, href } = getNavigationHeader();

  return (
    <Link href={href} className="h-fit flex items-center gap-1">
      <BoxIcon icon={<ArrowRight />} customClass="fill-secondary" />
      <Text variant={TEXT_VARIANT.SECONDARY} className="ml-1">
        {`Back to ${title}`}
      </Text>
    </Link>
  );
};

export default BackNavigation;
