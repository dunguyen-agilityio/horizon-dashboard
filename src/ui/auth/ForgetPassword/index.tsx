'use client';

import { useState } from 'react';

// Components
import ConfirmEmailForm from './ConfirmEmailForm';
import ResendEmailConfirm from './ResendEmailConfirm';

const ForgetPasswordForm = () => {
  const [email, setEmail] = useState('');

  const handleSendEmailConfirm = async (email: string) => {
    setEmail(email);
  };

  const renderContent = () => {
    switch (true) {
      case !email:
        return <ConfirmEmailForm onSendEmailConfirm={handleSendEmailConfirm} />;
      default:
        return <ResendEmailConfirm email={email} />;
    }
  };

  return <div className="w-full sm:w-96">{renderContent()}</div>;
};

export default ForgetPasswordForm;
