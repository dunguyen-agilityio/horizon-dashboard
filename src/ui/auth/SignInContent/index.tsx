import { Controller, useForm } from 'react-hook-form';

// Components
import { Link } from '@nextui-org/link';
import { Input } from '@nextui-org/input';
import { Checkbox } from '@nextui-org/checkbox';
import { Button, InputPassword, Text } from '@/components';

// Types
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';
import type { UserSignin } from '@/types/authentication';

// Utils
import { isEnableSubmitButton } from '@/utils/validation';

// Schema
import { useMemo } from 'react';

const REQUIRED_FIELDS = ['email', 'password'];

const SignInContent = () => {
  const signinFormInitValues: UserSignin = {
    email: '',
    password: '',
  };

  const {
    register,
    control,
    formState: { dirtyFields, errors },
  } = useForm<UserSignin>({
    mode: 'onBlur',
    reValidateMode: 'onBlur',
    values: signinFormInitValues,
  });

  const dirtyItems = Object.keys(dirtyFields);
  const isDisabled = useMemo(
    () => !isEnableSubmitButton(REQUIRED_FIELDS, dirtyItems, errors),
    [dirtyItems, errors],
  );

  return (
    <>
      <div className="pt-3 mb-8 flex flex-col gap-6">
        <Controller
          control={control}
          {...register('email', { required: 'Email is required' })}
          render={({
            field: { onChange, value, ...rest },
            fieldState: { error },
          }) => (
            <Input
              isRequired
              size="lg"
              key="email"
              type="email"
              label="Email"
              labelPlacement="outside"
              placeholder="mail@simmmple.com"
              value={value}
              onChange={(value) => {
                onChange(value);
              }}
              isInvalid={!!errors.email}
              errorMessage={error?.message}
              {...rest}
            />
          )}
        />
        <Controller
          control={control}
          {...register('password', { required: 'Password is required' })}
          render={({
            field: { onChange, value, ...rest },
            fieldState: { error },
          }) => (
            <InputPassword
              value={value}
              onChange={(value) => {
                onChange(value);
              }}
              isInvalid={!!errors.password}
              errorMessage={error?.message}
              {...rest}
            />
          )}
        />
      </div>

      <div className="flex justify-between">
        <Checkbox
          aria-label="checkbox-keep-login"
          defaultSelected
          radius="sm"
          className="fill-blue-450"
        >
          <Text as="p" size={TEXT_SIZE['md']}>
            Keep me logged in
          </Text>
        </Checkbox>
        <Link href="/forget-password">
          <Text
            size={TEXT_SIZE.sm}
            variant={TEXT_VARIANT.QUATERNARY}
            className="pr-3"
          >
            Forget password?
          </Text>
        </Link>
      </div>

      <Button
        isDisabled={isDisabled}
        className="bg-blue-450 dark:bg-purple-750 w-full py-7 mb-6 mt-8"
      >
        <Text size={TEXT_SIZE.sm} className="text-white font-bold leading-4">
          Sign In
        </Text>
      </Button>
    </>
  );
};

export default SignInContent;
