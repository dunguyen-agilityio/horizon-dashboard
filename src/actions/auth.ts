'use server';

// Libs
import { cookies } from 'next/headers';

// Config
import { signIn, signOut } from '@/auth.config';

// Constants
import { API_ENTITY } from '@/constants/api';
import { API_TOKEN, AUTH_ROUTES } from '@/constants';

// Types
import { ForgetResponse, SignInFormData, SignInResponse } from '@/types/auth';

// Services
import { apiClient } from '@/services/api';

/**
 *
 * @param formData SignInFormData
 * @returns hasError boolean
 */
export const handleSignIn = async (formData: SignInFormData) =>
  await signIn('credentials', { ...formData, redirect: false });

export const handleSignOut = async () => {
  await signOut({ redirectTo: AUTH_ROUTES.SIGN_IN });
  cookies().delete(API_TOKEN);
};

export const handleForgetPassword = async (email: string) =>
  await apiClient.post<ForgetResponse>(API_ENTITY.FORGET_PASSWORD, {
    body: { email },
  });

export type ResetPasswordPayload = {
  code: string;
  password: string;
  passwordConfirmation: string;
};

export const handleResetPassword = async (payload: ResetPasswordPayload) =>
  await apiClient.post<SignInResponse>(API_ENTITY.RESET_PASSWORD, {
    body: payload,
  });
