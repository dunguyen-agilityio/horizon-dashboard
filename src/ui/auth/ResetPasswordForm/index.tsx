'use client';

// Libs
import { useRouter } from 'next/navigation';
import { Controller, useForm } from 'react-hook-form';

// Components
import { Button, InputPassword, Text } from '@/components';

// Types
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';

// Constants
import { AUTH_ROUTES, MESSAGES, REGEX_PASSWORD } from '@/constants';

// Actions
import { handleResetPassword } from '@/actions/auth';

// Hooks
import useToast from '@/contexts/toast';

interface ResetPasswordFormData {
  password: string;
  passwordConfirmation: string;
}

const resetPasswordInitValues: ResetPasswordFormData = {
  password: '',
  passwordConfirmation: '',
};

interface ResetPasswordFormProps {
  code: string;
}

const { PASSWORD, RESET_PASSWORD } = MESSAGES;

const ResetPasswordForm = ({ code }: ResetPasswordFormProps) => {
  const {
    control,
    getValues,
    handleSubmit,
    formState: { isDirty },
  } = useForm<ResetPasswordFormData>({
    mode: 'onBlur',
    values: resetPasswordInitValues,
  });

  const { showToast } = useToast();
  const { push } = useRouter();

  const isDisabled = !isDirty;

  const resetPassword = async (data: ResetPasswordFormData) => {
    const { data: res } = await handleResetPassword({ code, ...data });
    const { ERROR, SUCCESS } = RESET_PASSWORD;

    const isSuccess = !!res?.user;

    const { TITLE, DESCRIPTION } = isSuccess ? SUCCESS : ERROR;

    showToast({
      title: TITLE,
      message: DESCRIPTION,
      type: isSuccess ? 'success' : 'error',
    });

    isSuccess && push(AUTH_ROUTES.SIGN_IN);
  };

  return (
    <form onSubmit={handleSubmit(resetPassword)} className="w-full sm:w-96">
      <div className="flex flex-col gap-6">
        <div className="flex flex-col gap-2 pb-8">
          <Text as="h1" size={TEXT_SIZE['2xl']} className="leading-[56px]">
            Reset your password
          </Text>
          <Text
            as="h2"
            size={TEXT_SIZE['md']}
            variant={TEXT_VARIANT.TERTIARY}
            className="leading-6"
          >
            Enter your new password to reset your password.
          </Text>
        </div>
        <Controller
          name="password"
          control={control}
          rules={{
            required: PASSWORD.REQUIRED,
            pattern: {
              value: REGEX_PASSWORD,
              message: PASSWORD.REGEX_MISMATCH,
            },
          }}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              isInvalid={invalid}
              errorMessage={error?.message}
              label="New Password"
              placeholder="New Password"
              {...field}
            />
          )}
        />
        <Controller
          name="passwordConfirmation"
          control={control}
          rules={{
            required: PASSWORD.REQUIRED,
            validate: (value) =>
              value === getValues('password') ? true : PASSWORD.MISMATCH,
          }}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              isInvalid={invalid}
              errorMessage={error?.message}
              label="Confirm Password"
              placeholder="Confirm Password"
              {...field}
            />
          )}
        />
        <Button
          isDisabled={isDisabled}
          className="w-full py-7 mt-6"
          type="submit"
          data-testid="update-password-btn"
        >
          <Text size={TEXT_SIZE.sm} className="text-white font-bold leading-4">
            Update Password
          </Text>
        </Button>
      </div>
    </form>
  );
};

export default ResetPasswordForm;
