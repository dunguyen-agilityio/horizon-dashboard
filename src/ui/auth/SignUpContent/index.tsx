'use client';

import { useState } from 'react';
import { useRouter } from 'next/navigation';
import { Controller, useForm } from 'react-hook-form';

// Components
import { Input } from '@nextui-org/input';
import { Button, InputPassword, Text } from '@/components';

// Types
import { TEXT_SIZE } from '@/types/text';
import type { SignUpFormData, SignUpPayload } from '@/types/auth';

// Regex
import { REGEX_EMAIL, REGEX_PASSWORD } from '@/constants/regex';

// Constants
import { AUTH_ROUTES } from '@/constants';

// Utils
import {
  emailRules,
  identifierRules,
  passwordRules,
  validateConfirmPassword,
} from '@/utils/validation';

// Actions
import { handleSignUp } from '@/actions/auth';

// Hooks
import useToast from '@/contexts/toast';

// Style
import { STYLE_INPUT } from '@/styles/input';

const signUpFormInitValues: SignUpFormData = {
  username: '',
  email: '',
  password: '',
  confirmPassword: '',
};

const SignUpContent = () => {
  const {
    watch,
    control,
    formState: { isDirty },
    handleSubmit,
  } = useForm<SignUpFormData>({
    mode: 'onBlur',
    values: signUpFormInitValues,
  });

  const [isLoading, setIsLoading] = useState(false);
  const isDisabled = !isDirty;
  const passwordValue = watch('password');

  const { replace } = useRouter();
  const { showToast } = useToast();

  const signUp = async ({ email, password, username }: SignUpPayload) => {
    setIsLoading(true);
    const { error } = await handleSignUp({ email, password, username });

    if (error) {
      showToast({
        title: 'Failed',
        message: error.message,
        type: 'error',
        timeOut: 5000,
      });
      setIsLoading(false);
      return;
    }

    replace(AUTH_ROUTES.SIGN_IN);
  };

  return (
    <form onSubmit={handleSubmit(signUp)} className="relative">
      <div className="pt-3 flex flex-col gap-6">
        <Controller
          name="email"
          control={control}
          rules={emailRules(REGEX_EMAIL)}
          render={({ field, fieldState: { error, invalid } }) => (
            <Input
              isRequired
              size="lg"
              label="Email"
              labelPlacement="outside"
              placeholder="Enter your email"
              isInvalid={invalid}
              classNames={STYLE_INPUT}
              errorMessage={error?.message}
              {...field}
            />
          )}
        />
        <Controller
          name="username"
          control={control}
          rules={identifierRules}
          render={({ field, fieldState: { error, invalid } }) => (
            <Input
              isRequired
              size="lg"
              label="User name"
              labelPlacement="outside"
              classNames={STYLE_INPUT}
              placeholder="Enter your user name"
              isInvalid={invalid}
              errorMessage={error?.message}
              {...field}
            />
          )}
        />
        <Controller
          name="password"
          control={control}
          rules={passwordRules(REGEX_PASSWORD)}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              isInvalid={invalid}
              classNames={STYLE_INPUT}
              errorMessage={error?.message}
              {...field}
            />
          )}
        />
        <Controller
          name="confirmPassword"
          control={control}
          rules={{
            required: 'This field is required',
            validate: (confirmPassword: string) =>
              validateConfirmPassword(passwordValue, confirmPassword),
          }}
          render={({ field, fieldState: { error, invalid } }) => (
            <InputPassword
              label="Confirm Password"
              isInvalid={invalid}
              classNames={STYLE_INPUT}
              errorMessage={error?.message}
              placeholder="Enter password confirm"
              {...field}
            />
          )}
        />
      </div>
      <Button
        isDisabled={isDisabled}
        isLoading={isLoading}
        className="bg-blue-450 dark:bg-purple-750 w-full py-7 mt-10"
        type="submit"
        data-testid="signup-btn"
      >
        {!isLoading && (
          <Text size={TEXT_SIZE.sm} className="text-white font-bold leading-4">
            Sign Up
          </Text>
        )}
      </Button>
      {isLoading && <div className="absolute inset-0" />}
    </form>
  );
};

export default SignUpContent;
