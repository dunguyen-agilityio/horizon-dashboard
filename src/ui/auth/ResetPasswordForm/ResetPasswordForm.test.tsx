import { fireEvent, render, waitFor } from '@/utils/test-utils';

import ResetPasswordForm from '.';

import { AUTH_ROUTES, MESSAGES } from '@/constants';
import { handleResetPassword } from '@/actions/auth';
import { MOCK_USERS } from '@/mocks/user';
import { useRouter } from 'next/navigation';

jest.mock('@/actions/auth', () => ({
  handleResetPassword: jest.fn(),
}));

let mockPush: jest.Mock;

const mockPassword = '12345@Abc';
const mockInvalidPassword = '1234';
const mockCode = '13232';

describe('ResetPassword Tests', () => {
  beforeEach(() => {
    mockPush = jest.fn();

    (useRouter as jest.Mock).mockImplementation(() => ({
      push: mockPush,
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should match snapshot', () => {
    const { container } = render(<ResetPasswordForm code={mockCode} />);
    expect(container).toMatchSnapshot();
  });

  it('Should render correctly', async () => {
    const { getByLabelText, queryByText } = render(
      <ResetPasswordForm code={mockCode} />,
    );

    const passwordField = getByLabelText('New Password');
    const confirmPasswordField = getByLabelText('Confirm Password');

    fireEvent.blur(passwordField);

    await waitFor(() => {
      expect(queryByText(MESSAGES.PASSWORD.REQUIRED)).toBeInTheDocument();
    });

    fireEvent.change(passwordField, { target: { value: mockInvalidPassword } });
    fireEvent.blur(passwordField);

    await waitFor(() => {
      expect(queryByText(MESSAGES.PASSWORD.REGEX_MISMATCH)).toBeInTheDocument();
    });

    fireEvent.change(passwordField, { target: { value: mockPassword } });
    fireEvent.blur(passwordField);
    fireEvent.change(confirmPasswordField, {
      target: { value: mockInvalidPassword },
    });
    fireEvent.blur(confirmPasswordField);

    await waitFor(() => {
      expect(queryByText(MESSAGES.PASSWORD.MISMATCH)).toBeInTheDocument();
    });

    fireEvent.change(confirmPasswordField, { target: { value: mockPassword } });
    fireEvent.blur(confirmPasswordField);

    await waitFor(() => {
      expect(queryByText(MESSAGES.PASSWORD.MISMATCH)).not.toBeInTheDocument();
    });
  });

  it('handleResetPassword and push will triggered correctly when fill on fields and click submit', async () => {
    (handleResetPassword as jest.Mock).mockResolvedValue({
      data: { user: MOCK_USERS[0] },
    });

    const { getByLabelText, getByTestId } = render(
      <ResetPasswordForm code={mockCode} />,
    );

    const passwordField = getByLabelText('New Password');
    const confirmPasswordField = getByLabelText('Confirm Password');

    fireEvent.change(passwordField, { target: { value: mockPassword } });
    fireEvent.blur(passwordField);
    fireEvent.change(confirmPasswordField, {
      target: { value: mockInvalidPassword },
    });

    fireEvent.change(confirmPasswordField, { target: { value: mockPassword } });
    fireEvent.blur(confirmPasswordField);

    fireEvent.submit(getByTestId('update-password-btn'));

    await waitFor(() => {
      expect(handleResetPassword).toHaveBeenNthCalledWith(
        1,
        expect.objectContaining({
          code: mockCode,
          password: mockPassword,
          passwordConfirmation: mockPassword,
        }),
      );

      expect(mockPush).toHaveBeenNthCalledWith(1, AUTH_ROUTES.SIGN_IN);
    });
  });

  it('push will not triggered when reset password not success', async () => {
    (handleResetPassword as jest.Mock).mockResolvedValue({
      data: { user: null },
    });

    const { getByLabelText, getByTestId } = render(
      <ResetPasswordForm code={mockCode} />,
    );

    const passwordField = getByLabelText('New Password');
    const confirmPasswordField = getByLabelText('Confirm Password');

    fireEvent.change(passwordField, { target: { value: mockPassword } });
    fireEvent.blur(passwordField);
    fireEvent.change(confirmPasswordField, {
      target: { value: mockInvalidPassword },
    });

    fireEvent.change(confirmPasswordField, { target: { value: mockPassword } });
    fireEvent.blur(confirmPasswordField);

    fireEvent.submit(getByTestId('update-password-btn'));

    await waitFor(() => {
      expect(mockPush).not.toHaveBeenCalled();
    });
  });
});
