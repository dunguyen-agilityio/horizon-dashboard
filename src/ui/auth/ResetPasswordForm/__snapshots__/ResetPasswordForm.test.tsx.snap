// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Tests Should match snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <form
      class="w-full sm:w-96 relative"
    >
      <div
        class="flex flex-col gap-6"
      >
        <div
          class="flex flex-col gap-2 pb-8"
        >
          <h1
            class="text-primary dark:text-white text-2xl font-bold leading-2xl leading-[56px]"
          >
            Reset your password
          </h1>
          <h2
            class="text-foreground dark:text-secondary text-medium leading-6"
          >
            Enter your new password to reset your password.
          </h2>
        </div>
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_12px)]"
          data-filled="true"
          data-filled-within="true"
          data-has-elements="true"
          data-has-label="true"
          data-required="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-12 min-h-12 rounded-large transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <label
                class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 after:content-['*'] after:text-danger after:ml-0.5 rtl:after:ml-[unset] rtl:after:mr-0.5 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 text-medium group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_24px)] pe-2 max-w-full text-ellipsis overflow-hidden"
                data-slot="label"
                for="react-aria-:r0:"
                id="react-aria-:r1:"
              >
                New Password
              </label>
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-label="inputPassword"
                  aria-labelledby="react-aria-:r0: react-aria-:r1:"
                  aria-required="true"
                  class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-medium group-data-[has-value=true]:text-default-foreground"
                  data-has-end-content="true"
                  data-slot="input"
                  data-testid="input-password"
                  id="react-aria-:r0:"
                  name="password"
                  placeholder="New Password"
                  type="password"
                  value=""
                />
                <button
                  aria-label="toggle password visibility"
                  class="focus:outline-none"
                  type="button"
                >
                  <i
                    class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary"
                  >
                    <svg
                      data-testid="eye-filled-icon"
                      height="24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5ZM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5Zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3Z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[calc(theme(fontSize.small)_+_12px)]"
          data-filled="true"
          data-filled-within="true"
          data-has-elements="true"
          data-has-label="true"
          data-required="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 bg-default-100 data-[hover=true]:bg-default-200 group-data-[focus=true]:bg-default-100 h-12 min-h-12 rounded-large transition-background motion-reduce:transition-none !duration-150 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <label
                class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block text-foreground-500 after:content-['*'] after:text-danger after:ml-0.5 rtl:after:ml-[unset] rtl:after:mr-0.5 will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] group-data-[filled-within=true]:text-foreground group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 top-1/2 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 text-medium group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_24px)] pe-2 max-w-full text-ellipsis overflow-hidden"
                data-slot="label"
                for="react-aria-:r5:"
                id="react-aria-:r6:"
              >
                Confirm Password
              </label>
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-label="inputPassword"
                  aria-labelledby="react-aria-:r5: react-aria-:r6:"
                  aria-required="true"
                  class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-medium group-data-[has-value=true]:text-default-foreground"
                  data-has-end-content="true"
                  data-slot="input"
                  data-testid="input-password"
                  id="react-aria-:r5:"
                  name="passwordConfirmation"
                  placeholder="Confirm Password"
                  type="password"
                  value=""
                />
                <button
                  aria-label="toggle password visibility"
                  class="focus:outline-none"
                  type="button"
                >
                  <i
                    class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary"
                  >
                    <svg
                      data-testid="eye-filled-icon"
                      height="24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5ZM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5Zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3Z"
                      />
                    </svg>
                  </i>
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary data-[hover=true]:opacity-hover text-black dark:text-white opacity-50 cursor-not-allowed w-full py-7 mt-6"
          data-disabled="true"
          data-testid="update-password-btn"
          disabled=""
          type="submit"
        >
          <p
            class="text-primary dark:text-white text-small font-bold leading-4"
          >
            Update Password
          </p>
        </button>
      </div>
    </form>
  </div>
</div>
`;
